cap-tempfile/src/tempfile.rs

```diff
@@ -100,17 +101,29 @@ fn generate_name_in(subdir: &Dir, f: &File) -> io::Result<String> {
 /// Create a new temporary file in the target directory, which may or may not
 /// have a (randomly generated) name at this point. If anonymous is specified,
 /// the file will be deleted
-fn new_tempfile(d: &Dir, anonymous: bool) -> io::Result<(File, Option<String>)> {
+fn new_tempfile(
+    d: &Dir,
+    anonymous: bool,
+    #[cfg(any(target_os = "android", target_os = "linux"))] unnamed_mode: Option<u32>,
+    #[cfg(unix)] named_mode: Option<u32>,
```

Comment:
The first condition implies the second, which makes things feel backwards to me.

This all might be cleaner if we used a `struct NewTempfileOpts` instead - then there'd be conditionals in way fewer places (just the definition and uses, not needing to thread them through all the internal APIs)

Reply:
> The first condition implies the second, which makes things feel backwards to me.

I can reorder, i think I put unnamed_mode first because I use it first

> This all might be cleaner if we used a `struct NewTempfileOpts` instead - then there'd be conditionals in way fewer places (just the definition and uses, not needing to thread them through all the internal APIs)

That would just change `new_tempfile` and `new_tempfile_linux` definition, and `new_tempfile_linux` doesn't need `named_mode`, so not much difference. I can do it in a separate commit if you want to see